src/controller/CHIPDeviceController.cpp

```diff
@@ -872,6 +904,59 @@ void DeviceCommissioner::OnDiscoveredDeviceOverBleError(void * appState, CHIP_ER
 }
 #endif // CONFIG_NETWORK_LAYER_BLE
 
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+void DeviceCommissioner::OnWiFiPAFSubscribeComplete(void * appState)
+{
+    std::weak_ptr<DeviceCommissioner> * caller     = (std::weak_ptr<DeviceCommissioner> *) (appState);
```

Comment:
But this is casting the pointer we passed (to shared_ptr) to a weak_ptr?

I don't understand how this works.